src/librustc_resolve/diagnostics.rs

```diff
@@ -444,6 +444,43 @@ impl SomeTrait for Foo { // ok!
 ```
 "##,
 
+E0406: r##"
+The function is referring to an associated type which hasn't been declared in
+the trait.
+
```

Comment:
You forgot "Example of erroneous code:". Take a look at the [RFC](https://github.com/rust-lang/rfcs/pull/1567) for more information.